### PR TITLE
cache index constraint on first parallel run (suggestion 2)

### PR DESCRIPTION
**Pull request by mauriyouth**

### CODE REVIEW

File: core/src/stores/sqlite.rs [hunk 1]
```diff
@@ -927,15 +927,21 @@ impl Store for SQLiteStore {
...
                 project_id,
                 created,
                 request.hash().to_string(),
                 request_data,
                 generation_data
-            ])?;
+            ]);
 
-            Ok(())
+            match statement {
```
Comment (spolu) [inline core/src/stores/sqlite.rs:938]: I think we can match the call to stmt.insert directly.

File: core/src/stores/sqlite.rs [hunk 2]
```diff
@@ -927,15 +927,21 @@ impl Store for SQLiteStore {
...
+                Ok(_) => Ok(()),
+                Err(e) => match e.to_string().as_str() {
```
Comment (spolu) [inline core/src/stores/sqlite.rs:940]: I thiiiiink there is an error type for that, matching on that would be nice?

**Review by spolu — DISMISSED** [review]
Generally LGTM! Would love to type match if possible but otherwise feel free to merge as is!

File: core/src/stores/sqlite.rs (hunk 2 above)
Comment (mauriyouth) [reply core/src/stores/sqlite.rs:938]: Yeah it's better but didn't find the corresponding error here https://docs.rs/rusqlite/0.7.2/rusqlite/enum.Error.html

Comment (mauriyouth) [reply core/src/stores/sqlite.rs:938]: moved the match upper 👍

**Review by spolu — APPROVED** [review]
LGTM!

**Comment by mauriyouth** [comment]
i dont have merge right, so you can merge it :)

**Comment by spolu** [comment]
🙌👊

**Pull request merged**